xdsl/dialects/arith.py

```diff
@@ -195,6 +196,29 @@ class SignlessIntegerBinaryOperation(IRDLOperation, abc.ABC):
 
     assembly_format = "$lhs `,` $rhs attr-dict `:` type($result)"
 
+    @staticmethod
+    def py_operation(lhs: int, rhs: int) -> int | None:
```

Comment:
Could you please add a simple doc string on what this does/is for? I find the name very unintuitive, although I understand it's hard (this looks like an `eval` as we have for affine maps?)

Reply:
ah wait a second could this be used in the interpreter also?

Reply:
Have added a docstring. Essentially this should return an integer when we know a canonicalization can happen. Unsure if this can/should be used in the interpreter